Forms/CartForm.cs

```diff
@@ -246,6 +246,19 @@ private void CheckoutButton_Click(object sender, EventArgs e)
                         _purchase.Date = DateTime.Now.ToString();
                         _purchase.Total = ComputeTotalPrice();
 
+                        foreach (PurchaseDetails purchaseDetails in _purchases)

```

Comment:
Optimize

Reply:
Will remove this. This part was already handled in customer factory.